src/charm.py

```diff
@@ -397,7 +406,8 @@ def _configure(self):  # noqa: C901
 
         # At this point we're already after the can_connect guard, so if the following pebble operations fail, better
         # to let the charm go into error state than setting blocked.
-        self._push_certs()
+        if self._cert_available_in_reldata and not self._tls_ready_on_disk:
```

Comment:
Apologies for the back and forth, but how about shortening:
- `_cert_available_in_reldata` -> `_certs_in_reldata`
- `_tls_ready_on_disk` -> `_certs_on_disk`